sql/semisync_master.cc

```diff
@@ -592,14 +585,14 @@ int Repl_semi_sync_master::report_reply_packet(uint32 server_id,
                           log_file_name, (ulong)log_file_pos, server_id));
 
   rpl_semi_sync_master_get_ack++;
-  report_reply_binlog(server_id, log_file_name, log_file_pos);
+  report_reply_binlog(thd, server_id, log_file_name, log_file_pos);
 
 l_end:
 
   DBUG_RETURN(result);
 }
 
-int Repl_semi_sync_master::report_reply_binlog(uint32 server_id,
+int Repl_semi_sync_master::report_reply_binlog(THD *thd, uint32 server_id,
```

Comment:
Perhaps rename `thd` to `slave_thd` (or something similar) to be specific that it is not the calling thread (ack receiver)

Reply:
Done.